internal/markup/markup.go

```diff
@@ -140,12 +140,22 @@ func RenderCrossReferenceIssueIndexPattern(rawBytes []byte, urlPrefix string, me
 }
 
 // RenderSha1CurrentPattern renders SHA1 strings to corresponding links that assumes in the same repository.
-func RenderSha1CurrentPattern(rawBytes []byte, urlPrefix string) []byte {
+func RenderSha1CurrentPattern(rawBytes []byte, urlPrefix string, metas map[string]string) []byte {
 	return []byte(Sha1CurrentPattern.ReplaceAllStringFunc(string(rawBytes[:]), func(m string) string {
 		if com.StrTo(m).MustInt() > 0 {
 			return m
 		}
-		return fmt.Sprintf(`<a href="%s/commit/%s"><code>%s</code></a>`, urlPrefix, m, tool.ShortSHA1(string(m)))
+
+		if len(metas) > 0 {
+			user := metas["user"]
+			repo := metas["repo"]
+
+			if user != "" && repo != "" {
+				urlPrefix = fmt.Sprintf("/%s/%s", user, repo)
+			}
+		}
```

Comment:
It is generally safe to assume `metas` won't be empty.

```suggestion
		return fmt.Sprintf(`<a href="%s/commit/%s"><code>%s</code></a>`, metas["repoLink"], m, tool.ShortSHA1(string(m)))
```